src/serde.rs

```diff
@@ -71,6 +71,42 @@ where
     }
 }
 
+/// Byte slice wrapper to serialize as a hex string in lowercase characters.
+pub struct SerializeBytesAsHex<'a>(pub(crate) &'a [u8]);
+
+impl<'a> serde::Serialize for SerializeBytesAsHex<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serialize_lower(self.0, serializer)
+    }
+}
+
+/// Byte slice wrapper to serialize as a hex string in lowercase characters.
+pub struct SerializeBytesAsHexLower<'a>(pub(crate) &'a [u8]);
+
+impl<'a> serde::Serialize for SerializeBytesAsHexLower<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serialize_lower(self.0, serializer)
+    }
+}
+
+/// Byte slice wrapper to serialize as a hex string in uppercase characters.
+pub struct SerializeBytesAsHexUpper<'a>(pub(crate) &'a [u8]);
+
+impl<'a> serde::Serialize for SerializeBytesAsHexUpper<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serialize_upper(self.0, serializer)
+    }
+}
+
```

Comment:
The indentation is missing.

Reply:
You sure? I can't see the problem neither can `rustfmt`?

Reply:
LOL, it's an optical illusion caused by GH UI :D

Reply:
Ha!